service/matching/taskListManager.go

```diff
@@ -359,6 +359,26 @@ func (c *taskListManagerImpl) CancelPoller(pollerID string) {
 
 // Returns a batch of tasks from persistence starting form current read level.
 func (c *taskListManagerImpl) getTaskBatch() ([]*persistence.TaskInfo, error) {
+	result := make([]*persistence.TaskInfo, 0)
+	readLevel := c.taskAckManager.getReadLevel()
+	maxReadLevel := c.taskWriter.GetMaxReadLevel()
+	// when gap is large, query db multiple times to avoid query aborted caused by tombstone
+	for readLevel < maxReadLevel {
```

Comment:
This loop is quite dangerous.  It could result in reading large number of tasks into memory.  I think we should short circuit the loop if GetTasks return with non-zero tasks.

Reply:
Make sense, addressed